diff --git a/arch/arm/dts/socfpga_stratix10_socdk.dts b/arch/arm/dts/socfpga_stratix10_socdk.dts
index 6622720..81d661e 100755
--- a/arch/arm/dts/socfpga_stratix10_socdk.dts
+++ b/arch/arm/dts/socfpga_stratix10_socdk.dts
@@ -9,7 +9,7 @@
 	model = "SoCFPGA Stratix 10 SoCDK";
 
 	aliases {
-		ethernet0 = &gmac0;
+		ethernet0 = &gmac2;
 		i2c0 = &i2c1;
 		serial0 = &uart0;
 	};
@@ -52,10 +52,10 @@
 	status = "okay";
 };
 
-&gmac0 {
+&gmac2 {
 	status = "okay";
 	phy-mode = "rgmii";
-	phy-handle = <&phy0>;
+	phy-handle = <&phy2>;
 
 	max-frame-size = <3800>;
 
@@ -63,7 +63,7 @@
 		#address-cells = <1>;
 		#size-cells = <0>;
 		compatible = "snps,dwmac-mdio";
-		phy0: ethernet-phy@0 {
+		phy2: ethernet-phy@2 {
 			reg = <4>;
 
 			txd0-skew-ps = <0>; /* -420ps */
@@ -91,7 +91,7 @@
 	cap-sd-highspeed;
 	cap-mmc-highspeed;
 	broken-cd;
-	bus-width = <4>;
+	bus-width = <8>;
 	drvsel = <3>;
 	smplsel = <0>;
 };
@@ -135,10 +135,6 @@
 	status = "okay";
 };
 
-&usb0 {
-	status = "okay";
-};
-
 &watchdog0 {
 	status = "okay";
 };
