diff --git a/recipes-bsp/device-tree/files/socfpga_agilex3_ghrd.dtsi b/recipes-bsp/device-tree/files/socfpga_agilex3_ghrd.dtsi
index 9e043ef..ceac3aa 100644
--- a/recipes-bsp/device-tree/files/socfpga_agilex3_ghrd.dtsi
+++ b/recipes-bsp/device-tree/files/socfpga_agilex3_ghrd.dtsi
@@ -49,6 +49,11 @@
 				resetvalue = <0>;
 		};
 */
+		jop@20020000 {
+			compatible = "generic-uio";
+			reg = <0x20020000 0x4000>;
+			reg-names = "jop";
+		};
 		soc_leds: leds {
 			compatible = "gpio-leds";
 
