diff --git a/recipes-bsp/device-tree/files/socfpga_ilc.dtsi b/recipes-bsp/device-tree/files/socfpga_ilc.dtsi
index 387d8bd..156dbb8 100644
--- a/recipes-bsp/device-tree/files/socfpga_ilc.dtsi
+++ b/recipes-bsp/device-tree/files/socfpga_ilc.dtsi
@@ -14,5 +14,10 @@
 			status = "disabled";
 			altr,sw-fifo-depth = <32>;
 		};
+		 jop@f9008000{
+                                compatible = "generic-uio";
+                                reg = <0xf9008000 0x5000>;
+                                reg-names = "jop";
+                };
 	};
 };
